### PR TITLE
Added support for AD4855

### PR DESCRIPTION
**Pull request by Shetty-Anush**
1. Support was added for AD4855 packet formats
2. Bit shift for setting channel offset was given appropriately

## Pull Request Description

Please replace this with a detailed description and motivation of the changes. 
You can tick the checkboxes below with an 'x' between square brackets or just check them after publishing the PR. 
If this PR contains a breaking change, list dependent PRs and try to push all related PRs at the same time.

## PR Type
- [ ] Bug fix (change that fixes an issue)
- [x] New feature (change that adds new functionality)
- [ ] Breaking change (has dependencies in other repos or will cause CI to fail)

## PR Checklist
- [x] I have followed the [Coding style guidelines](http://analogdevicesinc.github.io/no-OS/drivers_guide.html#coding-style)
- [x] I have performed a self-review of the changes
- [x] I have commented my code, at least hard-to-understand parts
- [ ] I have build all projects affected by the changes in this PR
- [x] I have tested in hardware affected projects, at the relevant boards
- [x] I have signed off all commits from this PR
- [ ] I have updated the documentation (wiki pages, ReadMe etc), if applies

### CODE REVIEW

File: drivers/adc/ad4858/ad4858.c [hunk 1]
```diff
@@ -481,8 +494,13 @@ int ad4858_set_chn_offset(struct ad4858_dev *dev, uint8_t chn, uint32_t offset)
 	if (!dev || (chn >= AD4858_NUM_CHANNELS))
 		return -EINVAL;
 
-	/* Offset value is spanned over bits [24:4] */
-	ret = ad4858_reg_write(dev, AD4858_REG_CH_OFFSET(chn), (offset << 4));
+	if (dev->prod_id == AD4855_PROD_ID_L)
+		/* Offset value is spanned over bits [23:8] */
```
Comment (buha) [inline drivers/adc/ad4858/ad4858.c:498]: this is a bug

```
if (condition)
    // comment
    do1();
else
    // comment
    do2();
```

it executes like this:
```
if (condition)
    // comment
do1();  // Always executes
else
    // comment
do2();  // Always executes
```

And this is not what you wanted. That's why MISRA enforces {} for all if/else statements even if they enclose oneliners. Put brackets and you'll be fine.

Comment (buha) [reply drivers/adc/ad4858/ad4858.c:498]: also, the compiler gave you warnings that you ignored, i'm pretty sure you'd get a warnings that the if/else does nothing

File: drivers/adc/ad4858/ad4858.c [hunk 2]
```diff
@@ -685,46 +707,83 @@ int ad4858_spi_data_read(struct ad4858_dev *dev, struct ad4858_conv_data *data)
...
+			/* 16-bit conversion result */
+			for (chn = 0; chn < AD4858_NUM_CHANNELS; chn++) {
+				indx = chn * 2;
+				data->raw[chn] = ((uint16_t)buff[indx] << 8) |
+						 ((uint16_t)buff[indx + 1]);
+			}
+			break;
+
+		case AD4858_PACKET_24_BIT:
+			/* 16-bit conversion result + 1-bit OR/UR + 3-bit channel ID + 4-bit softspan ID */
```
Comment (buha) [inline drivers/adc/ad4858/ad4858.c:723]: I would love to see more of NO_OS_BIT, NO_OS_GENMASK, no_os_field_prep, no_os_field_get being used. Some of these get pretty complex and hard to read. I think proper mask macros with proper names would help here. This comment applies to all the modifications not just this line.

**Review by buha — COMMENTED** [review]
also, thematically split up the commit in 2 commits:
1. Support was added for AD4855 packet formats
2. Bit shift for setting channel offset was given appropriately

File: drivers/adc/ad4858/ad4858.c (hunk 1 above)
Comment (Shetty-Anush) [reply drivers/adc/ad4858/ad4858.c:498]: It worked as expected on my end
I've added the brackets to make it unambiguous

**Review by buha — APPROVED** [review]

**Pull request merged**